### ISSUE TITLE
linter: `--type-aware` option immedaitely throws panic

### ISSUE THREAD

**Issue by KieranP** [oxc-project/oxc#12950]
### What version of Oxlint are you using?

1.11.1

### What command did you run?

oxlint --type-aware

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "ignorePatterns": ["/node_modules", "/dist", "/*.{js,ts}"],
  "plugins": ["eslint", "oxc", "typescript", "import", "promise", "unicorn"],
  "categories": {
    "correctness": "error",
    "nursery": "error",
    "pedantic": "error",
    "perf": "error",
    "restriction": "error",
    "style": "error",
    "suspicious": "error"
  }
}
```

### What happened?

Not sure you want errors for type-aware flag yet, given that it is still a WIP, but just in case, I upgrade from 0.10.1 to 0.11.0 and `oxlint` continues to work fine, but `oxlint --type-aware` throws the following:

```
› yarn oxlint --type-aware
panic: ScriptKind must be specified when parsing source files.

goroutine 232 [running]:
github.com/microsoft/typescript-go/internal/parser.(*Parser).initializeState(0x101d67c20?, {{0x140000ab400, 0x3c}, {0x14000264240, 0x3c}, {0x0, 0x0, 0x0, 0x0}, {0x1, ...}, ...}, ...)
	/home/runner/work/tsgolint/tsgolint/typescript-go/internal/parser/parser.go:210 +0x3a0
github.com/microsoft/typescript-go/internal/parser.ParseSourceFile({{0x140000ab400, 0x3c}, {0x14000264240, 0x3c}, {0x0, 0x0, 0x0, 0x0}, {0x1, 0x0}, ...}, ...)
	/home/runner/work/tsgolint/tsgolint/typescript-go/internal/parser/parser.go:113 +0xe8
github.com/microsoft/typescript-go/internal/project.(*DocumentRegistry).getParsedFile(0x140000f8b40, {{0x140000ab400, 0x3c}, {0x14000264240, 0x3c}, {0x0, 0x0, 0x0, 0x0}, {0x1, ...}, ...}, ...)
	/home/runner/work/tsgolint/tsgolint/typescript-go/internal/project/documentregistry.go:128 +0xdc
github.com/microsoft/typescript-go/internal/project.(*DocumentRegistry).getDocumentWorker(0x140000f8b40, 0x14000787040, {{{0x140000ab400, 0x3c}, {0x14000264240, 0x3c}, {0x0, 0x0, 0x0, 0x0}, ...}, ...})
	/home/runner/work/tsgolint/tsgolint/typescript-go/internal/project/documentregistry.go:101 +0x27c
github.com/microsoft/typescript-go/internal/project.(*DocumentRegistry).AcquireDocument(0x140000f8b40, 0x140000ab400?, {{0x140000ab400, 0x3c}, {0x14000264240, 0x3c}, {0x0, 0x0, 0x0, 0x0}, ...}, ...)
	/home/runner/work/tsgolint/tsgolint/typescript-go/internal/project/documentregistry.go:58 +0xec
github.com/microsoft/typescript-go/internal/project.(*Project).GetSourceFile(0x140003b5980, {{0x140000ab400, 0x3c}, {0x14000264240, 0x3c}, {0x0, 0x0, 0x0, 0x0}, {0x1, ...}, ...})
	/home/runner/work/tsgolint/tsgolint/typescript-go/internal/project/project.go:288 +0xfc
github.com/microsoft/typescript-go/internal/compiler.(*fileLoader).parseSourceFile(0x14002241108, 0x14000235440)
	/home/runner/work/tsgolint/tsgolint/typescript-go/internal/compiler/fileloader.go:367 +0x180
github.com/microsoft/typescript-go/internal/compiler.(*parseTask).load(0x14000235440, 0x14002241108)
	/home/runner/work/tsgolint/tsgolint/typescript-go/internal/compiler/parsetask.go:63 +0x168
github.com/microsoft/typescript-go/internal/compiler.(*fileLoaderWorker[...]).start.func1()
	/home/runner/work/tsgolint/tsgolint/typescript-go/internal/compiler/fileloadertask.go:84 +0x1e8
github.com/microsoft/typescript-go/internal/core.(*parallelWorkGroup).Queue.func1()
	/home/runner/work/tsgolint/tsgolint/typescript-go/internal/core/workgroup.go:39 +0x5c
created by github.com/microsoft/typescript-go/internal/core.(*parallelWorkGroup).Queue in goroutine 1
	/home/runner/work/tsgolint/tsgolint/typescript-go/internal/core/workgroup.go:37 +0x84
```

I tried to get more info for you about which file is causing the issue, but there isn't a `--debug` or `--verbose` flag. Also, it's odd that the error refers to `/home/runner/work` when I'm on a mac, and that path does not exist. Something mistakenly hardcoded?

**Comment by Sysix** [comment]
Could you please check if this is fixed in 1.11.2? :)

**Comment by KieranP** [comment]
@Sysix Yup, appears to be doing something now.

Bit of an increase in runtime though (23x):

```
› yarn lint:oxlint
Found 0 warnings and 0 errors.
Finished in 16ms on 13 files using 12 threads.

› yarn lint:oxlint --type-aware
Found 0 warnings and 0 errors.
Finished in 369ms on 13 files using 12 threads.
```

But I guess that is to be expected, and it's still far less than eslint-typescript.

**Comment by camc314** [comment]
> Bit of an increase in runtime though (23x):

we are working on perf here > https://github.com/oxc-project/tsgolint/issues/67
